tools/webhook_test.py

```diff
@@ -125,6 +145,46 @@ def int_or_default(input_parm):
         raw_input(), payload["message"][input_parm])
 
 
+def get_gym_info(gym_id):
+    """ Gets the information about the gym. """
+    return _gym_info.get(gym_id, _default_gym_info)
+
+
+def check_gym(questionable_input):
+    if get_gym_info(questionable_input)['name'] != "unknown":
+        print "Gym found = {}".format(get_gym_info(questionable_input)['name'])
+        return questionable_input, get_gym_info(questionable_input)['name']
+    else:
+        print "Not a valid gym. Please try again.."
+        check_gym(raw_input())
+
+
+def gym_or_invalid(prm, prm2):
+    payload["message"][prm], payload["message"][prm2] = check_gym(raw_input())
+
+
+def cache_or_invalid():
+    path = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    input = raw_input()
+    if os.path.exists(os.path.join(path, "cache", "{}.cache".format(input))):
```

Comment:
It would probably also be better to present the managers found in a list presented to the user and allowing them to pick one.